site/app/libraries/socket/Server.php

```diff
@@ -17,32 +17,40 @@ class Server implements MessageComponentInterface {
     // Holds the mapping between Connection Objects (key) and User_ID (value)
     private $sessions;
 
-    // Holds the mapping between Connection Objects IDs (key) and user current course (value)
-    private $courses;
+    // Holds the mapping between Connection Objects IDs (key) and user current course/page (value)
+    private $pages;
 
     // Holds the mapping between User_ID (key) and Connection object (value)
-    /** @var array<string, \Ratchet\ConnectionInterface> */
+    /** @var array<string, ConnectionInterface> */
     private $users;
 
     /** @var Core */
     private $core;
 
     public function __construct(Core $core) {
         $this->clients = [];
-        $this->courses = [];
+        $this->pages = [];
         $this->sessions = [];
-
+        $this->users = [];
         $this->core = $core;
     }
 
     /**
      * This function checks if a given connection object is authenticated
      * It uses the submitty_session cookie in the header data to work
+     * @param ConnectionInterface $conn
+     * @return bool
      */
     private function checkAuth(ConnectionInterface $conn): bool {
         $request = $conn->httpRequest;
-        $client_id = $conn->resourceId;
-        $origin = $request->getHeader('origin')[0];
+        $user_agent = $request->getHeader('User-Agent')[0];
+
+        if ($user_agent === 'websocket-client-php') {
+            $session_secret = $request->getHeader('Session-Secret')[0];
+            if ($session_secret  === $this->core->getConfig()->getSecretSession()) {
+                return true;
+            }
+        }
 
         $cookieString = $request->getHeader("cookie")[0];
         parse_str(strtr($cookieString, ['&' => '%26', '+' => '%2B', ';' => '&']), $cookies);
```

Comment:
Would it be appropriate to add an `else { return false; }` here?  I imagine the php-client doesn't have a cookie set and I'm just wondering if its possible we might end up trying to run functions on null values and result in an exception.

Reply:
yeah, makes sense